File: templates/rust/part1.rs
use std::io::{self, Read};

mod common;
use common::parse_input;

fn solve(input: &str) -> i64 {
    let _lines = parse_input(input);
    // TODO: Implement solution
    0
}

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let result = solve(&input);
    println!("{}", result);
}

File: templates/typescript/part1.ts
#!/usr/bin/env ts-node
/** Advent of Code - Part 1 */

import * as fs from 'fs';
import { parseInput } from './common';

function solve(data: string): number {
    const lines = parseInput(data);
    // TODO: Implement solution
    return 0;
}

// Read from stdin
const STDIN_FD = 0; // File descriptor for stdin
const data = fs.readFileSync(STDIN_FD, 'utf-8').trim();
const result = solve(data);
console.log(result);
